volplugin/handlers.go

```diff
@@ -260,8 +260,16 @@ func (dc *DaemonConfig) mount(w http.ResponseWriter, r *http.Request) {
 	}
 
 	volName := volConfig.String()
+	mountCount:= dc.increaseMount(volName)
 
-	if dc.increaseMount(volName) > 1 {
+	if !volConfig.Unlocked && mountCount > 1 {
+		if mountCount > 2 { // keeping counter=2 during locked mounts helps docker unmount from removing the actual mount
```

Comment:
Good catch. You are right. It is not required and In fact that code path will never be executed as consecutive unmount after mount failure will always keep counter at 1 and never > 2. I'm working on fixing this.